src/modules/FunModule.h

```diff
@@ -40,11 +40,27 @@ class FunModule : public Module, public QObject
 		PollSettings(const PollOptions& op, long long maxvotes, snowflake_t chan, const QString& name, int num, double time, const ServerPolls& pollList, snowflake_t msg);
 	};
 	Polls m_polls;
-	
+
+	enum GithubFlag
+	{
+		GITHUB_RANDOM,
+		GITHUB_REPOSITORY,
+		GITHUB_SEARCH
+	};
+
+	struct GithubRepo
+	{
+		QString fullname;
+		QString language;
+		int stars;
+		QString url;
+	};
+
 	//! A list of roles (for each server) that have been given poll 
 	//! creation access
 	QMap<snowflake_t, QList<snowflake_t>> m_pollWhitelist;
 	QNetworkAccessManager m_MemeManager, m_GithubManager;
+	GithubFlag m_GithubFlag = GITHUB_RANDOM;
```

Comment:
This probably shouldn't be be saved as a member variable. What happens if two !github requests are made at once, each with different flags?